common/persistence/visibility/store/elasticsearch/query_interceptors.go

```diff
@@ -66,29 +76,49 @@ func NewValuesInterceptor(
 	namespaceName namespace.Name,
 	saTypeMap searchattribute.NameTypeMap,
 	searchAttributesMapperProvider searchattribute.MapperProvider,
+	nameInterceptor *nameInterceptor,
 ) *valuesInterceptor {
 	return &valuesInterceptor{
 		namespace:                      namespaceName,
 		searchAttributesTypeMap:        saTypeMap,
 		searchAttributesMapperProvider: searchAttributesMapperProvider,
+		nameInterceptor:                nameInterceptor,
 	}
 }
 
 func (ni *nameInterceptor) Name(name string, usage query.FieldNameUsage) (string, error) {
 	fieldName := name
-	if searchattribute.IsMappable(name) {
+	if strings.EqualFold(fieldName, searchattribute.ScheduleID) {
```

Comment:
are other search attributes case-insensitive? I thought they weren't

Reply:
@rodrigozhou -- do you know the expected behavior? Generally speaking, I think it would be wise for SAs to be case-insensitive from the user's perspective, or at least normalize them at the server input boundary.

Reply:
Search attributes are case-sensitive.